interface/templates/interface/homepage.html

```diff
@@ -18,7 +18,7 @@
       {% for assignment in assignments %}
         <li>
           {{ assignment.name }}:
-          {% if assignment.is_active %}
+          {% if assignment.is_active and assignment.assignment_on or assignment.is_active and user in assignment.course.teaching_assistants.all %}
```

Comment:
It might not be enough just to hide the upload button. Students could still find the url to the upload page for this assignment and use it. A check in the upload view is still necessary